meilisearch/src/error.rs

```diff
@@ -98,14 +98,29 @@ impl From<MeilisearchHttpError> for aweb::Error {
 
 impl From<aweb::error::PayloadError> for MeilisearchHttpError {
     fn from(error: aweb::error::PayloadError) -> Self {
-        MeilisearchHttpError::Payload(PayloadError::Payload(error))
+        match error {
+            aweb::error::PayloadError::Incomplete(_) => MeilisearchHttpError::Payload(
+                PayloadError::Payload(ActixPayloadError::IncompleteError),
+            ),
+            _ => MeilisearchHttpError::Payload(PayloadError::Payload(
+                ActixPayloadError::OtherError(error),
+            )),
+        }
     }
 }
 
+#[derive(Debug, thiserror::Error)]
+pub enum ActixPayloadError {
+    #[error("The provided payload is incomplete and cannot be decompressed")]
+    IncompleteError,
```

Comment:
Is this error only happening for compressed payloads?

Reply:
Hey @irevoire,
Thanks for pointing this out,
I have gone through Actix web docs for finding any references,
I don't have any concrete evidence that it will only be thrown in decompression.

Can we add a generalised error message

"The provided payload is incomplete" or 
"The provided payload is incomplete and cannot be parsed"

Reply:
>  "The provided payload is incomplete and cannot be parsed"

That seems better than the original one yes

Reply:
Updated the message and also added the test for it,
Please review and let me know of any more changes